builtin/logical/database/path_roles_test.go

```diff
@@ -1352,6 +1353,56 @@ func TestIsInsideRotationWindow(t *testing.T) {
 	}
 }
 
+// TestStaticRoleNextVaultRotationOnPopulate tests the case where
+// a static role was created before the 1.15.0 release and the
+// NextVaultRotation was not set. It ensures that the NextVaultRotation
+// is set to the next rotation time when the role is read from storage
+// and the queue is populated.
+func TestStaticRoleNextVaultRotationOnPopulate(t *testing.T) {
+	ctx := context.Background()
+	b, storage, mockDB := getBackend(t)
+	defer b.Cleanup(ctx)
+	configureDBMount(t, storage)
+
+	roleName := "hashicorp"
+	data := map[string]interface{}{
+		"username":        "hashicorp",
+		"db_name":         "mockv5",
+		"rotation_period": "10m",
+	}
+
+	createStaticRoleWithData(t, b, storage, mockDB, roleName, data)
+	item, err := b.credRotationQueue.Pop()
+	require.NoError(t, err)
+	firstPriority := item.Priority
+
+	// force NextVaultRotation to zero to simulate roles before 1.15.0
+	role, err := b.StaticRole(context.Background(), storage, roleName)
+	require.NoError(t, err)
+	role.StaticAccount.NextVaultRotation = time.Time{}
+	entry, err := logical.StorageEntryJSON(databaseStaticRolePath+roleName, role)
+	require.NoError(t, err)
+	if err := storage.Put(context.Background(), entry); err != nil {
+		t.Fatal("failed to write role to storage", err)
+	}
+
+	// Confirm that NextVaultRotation is nil
```

Comment:
nit: it looks like we are asserting that `NextVaultRotation` is `time.Time{}` rather than nil.

Reply:
Thanks for calling that out. I've been using nil, when I really mean the zero value of time.Time. I'll replace those!